index.js

```diff
@@ -1,30 +1,194 @@
-main(input, 30);
+
+const generateFromQueue = async () => {
+    let data;
+    try {
+        data = await getMessage();
+    } catch (err) {
+        console.error('Unable to get message from queue:', err);
+        setTimeout(generateFromQueue, 10000);
+        return;
+    }
+
+    if (!data.Messages) {
+        console.error('No messages in queue');
+        generateFromQueue();
```

Comment:
this looks very confusing. i think you try to always wait for message. maybe create a function
consume(msgHandler, errHandler) or something like that, and make this consume function looping

Reply:
I think that's a great suggestion! I will try implementing that, thanks!

Reply:
see suggestion above